### PR TITLE
[cli] [test] Add some useful pytest arguments

### PR DESCRIPTION
**Pull request by archibate**
Related issue = #

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
e.g.:
`ti test bitmasked -k huge_bitmasked`
will only run the `test_bitmasked.py::test_huge_bitmasked`.

`ti test -x`
will exit immediately after first failure occured.

### CODE REVIEW

File: python/taichi/main.py [hunk 1]
```diff
@@ -815,6 +823,34 @@ def test(self, arguments: list = sys.argv[2:]):
                             dest='rerun',
                             type=str,
                             help='Rerun failed tests for given times')
+        parser.add_argument('-k',
+                            '--keys',
+                            required=False,
+                            default=None,
+                            dest='keys',
+                            type=str,
+                            help='Only run tests that matched the keys')
```
Comment (xumingkuan) [inline python/taichi/main.py:832]:
```suggestion
                            help='Only run tests that match the keys')
```

File: python/taichi/main.py [hunk 2]
```diff
@@ -815,6 +823,34 @@ def test(self, arguments: list = sys.argv[2:]):
...
+        parser.add_argument('-m',
+                            '--marks',
+                            required=False,
+                            default=None,
+                            dest='marks',
+                            type=str,
+                            help='Only run tests with specific marks')
```
Comment (xumingkuan) [inline python/taichi/main.py:839]: What're marks?

File: python/taichi/main.py [hunk 3]
```diff
@@ -815,6 +823,34 @@ def test(self, arguments: list = sys.argv[2:]):
...
+                            dest='failed_first',
+                            action='store_true',
+                            help='Run the previously failed test first')
+        parser.add_argument('-x',
+                            '--fail-fast',
+                            required=False,
+                            default=None,
+                            dest='fail_fast',
+                            action='store_true',
+                            help='Exit instantly on first failed test')
```
Comment (xumingkuan) [inline python/taichi/main.py:853]:
```suggestion
                            help='Exit instantly on the first failed test')
```

File: python/taichi/main.py (hunk 2 above)
Comment (archibate) [reply python/taichi/main.py:839]: e.g.:
```
@pytest.mark.xumingkuan
def test_hello():
  ...
```
```bash
ti test -m xumingkuan
```
I'll add a doc later clarify the test infrastructure.

**Pull request merged**